lib/routes/shared/security_pin/backup_phrase/verify_backup_phrase_page.dart

```diff
@@ -0,0 +1,220 @@
+import 'package:breez/bloc/backup/backup_bloc.dart';
+import 'package:breez/bloc/blocs_provider.dart';
+import 'package:breez/bloc/user_profile/breez_user_model.dart';
+import 'package:breez/bloc/user_profile/security_model.dart';
+import 'package:breez/bloc/user_profile/user_actions.dart';
+import 'package:breez/bloc/user_profile/user_profile_bloc.dart';
+import 'package:breez/routes/shared/backup_in_progress_dialog.dart';
+import 'package:breez/routes/shared/security_pin/backup_phrase/generate_backup_phrase_page.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+import 'package:breez/widgets/error_dialog.dart';
+import 'package:breez/widgets/route.dart';
+import 'package:flutter/material.dart';
+
+class VerifyBackupPhrasePage extends StatefulWidget {
+  final String _mnemonics;
+
+  VerifyBackupPhrasePage(
+    this._mnemonics,
+  );
+
+  @override
+  VerifyBackupPhrasePageState createState() => new VerifyBackupPhrasePageState();
+}
+
+class VerifyBackupPhrasePageState extends State<VerifyBackupPhrasePage> {
+  final _formKey = GlobalKey<FormState>();
+  String _title;
+  List _randomlySelectedIndexes = List();
+  List<String> _verificationFormValues;
+  List<String> _mnemonicsList;
+  bool _hasError;
+
+  @override
+  void initState() {
+    _verificationFormValues = List();
+    _mnemonicsList = widget._mnemonics.split(" ");
+    _title = "Let's verify";
+    _hasError = false;
+    _selectIndexes();
+    super.initState();
+  }
+
+  _selectIndexes() {
+    List list = List.generate(23, (i) => i);
+    list.shuffle();
+    for (var i = 0; i < 3; i++) {
+      _randomlySelectedIndexes.add(list[i]);
+    }
+    _randomlySelectedIndexes.sort();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    UserProfileBloc userProfileBloc = AppBlocsProvider.of<UserProfileBloc>(context);
+    BackupBloc backupBloc = AppBlocsProvider.of<BackupBloc>(context);
+    return WillPopScope(
```

Comment:
This way seems a bit odd and I think the requirements were misunderstood.
When user goes back from "Let's Verify" we need to show the previous screens as is, not regenerate the mnemonics at this stage.
If the user then enters the "Let's Verify" again (and each time he enters this screen) we need to pick newly random words.

Reply:
Reverted the changes [here](https://github.com/breez/breezmobile/pull/173/commits/87f74be999f3fb63ef0f74eaa9a93f858acd5498)
_(There's minor formatting changes)_